From a9608536a96de4a195261a1879d006e38bdf0cca Mon Sep 17 00:00:00 2001
From: Garvita Jain <garvitajain@google.com>
Date: Thu, 6 Feb 2025 07:29:08 +0000
Subject: [PATCH 1/2] Migrate Storage settings pages

This change includes -
* Using SettingsLib.CardPreferenceWidget in storage dashboard
* Using material theme color for progress bar

Test: manual
BUG: 349670985
Flag: com.android.settingslib.widget.theme.flags.is_expressive_design_enabled
Change-Id: Ib86ec9d1713316dd3dd3067c2cce87c89e15d9e0
---
 res/drawable/storage_progress_bar.xml  | 17 +++++++++++++++++
 res/layout/storage_item.xml            |  3 ++-
 res/xml/storage_dashboard_fragment.xml |  2 +-
 3 files changed, 20 insertions(+), 2 deletions(-)
 create mode 100644 res/drawable/storage_progress_bar.xml

diff --git a/res/drawable/storage_progress_bar.xml b/res/drawable/storage_progress_bar.xml
new file mode 100644
index 00000000000..3232230e7b4
--- /dev/null
+++ b/res/drawable/storage_progress_bar.xml
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:id="@android:id/background">
+        <shape>
+            <corners android:radius="8dp" />
+            <solid android:color="@color/settingslib_materialColorSurfaceContainerHighest" />
+        </shape>
+    </item>
+    <item android:id="@android:id/progress">
+        <clip>
+            <shape>
+                <corners android:radius="8dp" />
+                <solid android:color="@color/settingslib_materialColorPrimary" />
+            </shape>
+        </clip>
+    </item>
+</layer-list>
\ No newline at end of file
diff --git a/res/layout/storage_item.xml b/res/layout/storage_item.xml
index 3b58759baf0..98bedfe34dc 100644
--- a/res/layout/storage_item.xml
+++ b/res/layout/storage_item.xml
@@ -74,7 +74,8 @@
                 android:max="100"
                 android:layout_below="@android:id/title"
                 app:layout_constraintTop_toBottomOf="@android:id/title"
-                style="?android:attr/progressBarStyleHorizontal" />
+                style="?android:attr/progressBarStyleHorizontal"
+                android:progressDrawable="@drawable/storage_progress_bar" />
         </androidx.constraintlayout.widget.ConstraintLayout>
     </LinearLayout>
 </LinearLayout>
diff --git a/res/xml/storage_dashboard_fragment.xml b/res/xml/storage_dashboard_fragment.xml
index 6c2b8d48b66..92c5171ed80 100644
--- a/res/xml/storage_dashboard_fragment.xml
+++ b/res/xml/storage_dashboard_fragment.xml
@@ -38,7 +38,7 @@
         android:icon="@drawable/ic_storage"
         android:order="3"
         settings:controller="com.android.settings.deviceinfo.storage.AutomaticStorageManagementSwitchPreferenceController"/>
-    <com.android.settings.widget.CardPreference
+    <com.android.settingslib.widget.CardPreference
         android:key="free_up_space"
         android:order="4"
         android:title="@string/storage_free_up_space_title"
-- 
2.34.1


From 1fd7820dea53e6c7d868ad51d592d5c8e15cef49 Mon Sep 17 00:00:00 2001
From: Garvita Jain <garvitajain@google.com>
Date: Fri, 21 Feb 2025 11:02:31 +0000
Subject: [PATCH 2/2] Migrate Storage settings dashboard for Work profile

This change includes using SettingsLib.CardPreferenceWidget in storage
dashboard for devices with work profile enabled.

Test: manual
BUG: 349670985
Flag: com.android.settingslib.widget.theme.flags.is_expressive_design_enabled
Change-Id: Ib164df1f9d3507dd8d77d038210fe0eb0ca647c9
---
 res/xml/storage_category_fragment.xml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/res/xml/storage_category_fragment.xml b/res/xml/storage_category_fragment.xml
index 8af66396bd5..1742832df2a 100644
--- a/res/xml/storage_category_fragment.xml
+++ b/res/xml/storage_category_fragment.xml
@@ -19,7 +19,7 @@
     xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:settings="http://schemas.android.com/apk/res-auto"
     android:title="@string/storage_settings">
-    <com.android.settings.widget.CardPreference
+    <com.android.settingslib.widget.CardPreference
         android:key="free_up_space"
         android:order="4"
         android:title="@string/storage_free_up_space_title"
-- 
2.34.1

